internal/pkg/remote/client.go

```diff
@@ -155,45 +161,58 @@ func (c *Client) Send(
 	}
 
 	c.log.Debugf("Request method: %s, path: %s, reqID: %s", method, path, reqID)
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	requester := c.nextRequester()
+	c.clientMu.Lock()
+	defer c.clientMu.Unlock()
 
-	req, err := requester.request(method, path, params, body)
-	if err != nil {
-		return nil, errors.Wrapf(err, "fail to create HTTP request using method %s to %s", method, path)
-	}
+	var err error
+	var req *http.Request
+	var resp *http.Response
 
-	// Add generals headers to the request, we are dealing exclusively with JSON.
-	// Content-Type / Accepted type can be override from the called.
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Add("Accept", "application/json")
-	// This header should be specific to fleet-server or remove it
-	req.Header.Set("kbn-xsrf", "1") // Without this Kibana will refuse to answer the request.
+	c.sortClients()
+	for i, requester := range c.clients {
+		req, err = requester.newRequest(method, path, params, body)
+		if err != nil {
+			return nil, errors.Wrapf(err, "fail to create HTTP request using method %s to %s", method, path)
+		}
 
-	// If available, add the request id as an HTTP header
-	if reqID != "" {
-		req.Header.Add("X-Request-ID", reqID)
-	}
+		// Add generals headers to the request, we are dealing exclusively with JSON.
+		// Content-Type / Accepted type can be overridden by the caller.
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Add("Accept", "application/json")
+		// This header should be specific to fleet-server or remove it
+		req.Header.Set("kbn-xsrf", "1") // Without this Kibana will refuse to answer the request.
 
-	// copy headers.
-	for header, values := range headers {
-		for _, v := range values {
-			req.Header.Add(header, v)
+		// If available, add the request id as an HTTP header
+		if reqID != "" {
+			req.Header.Add("X-Request-ID", reqID)
+		}
+
+		// copy headers.
+		for header, values := range headers {
+			for _, v := range values {
+				req.Header.Add(header, v)
+			}
 		}
-	}
 
-	requester.lastUsed = time.Now().UTC()
+		requester.lastUsed = time.Now().UTC()
+
+		resp, err = requester.client.Do(req.WithContext(ctx))
+		if err != nil {
+			requester.lastErr = err
+			requester.lastErrOcc = time.Now().UTC()
+
+			// Using debug level as the error is only relevant if all clients fail.
+			c.log.With("error", err).Debugf("requester %d/%d to host %s errored",
```

Comment:
this approach is fine, we could also do multiple errors and return if all fails so we have context right away without a need to ask for debug logs.
last error may not be the one which moves us forward with SDH investigation